store/src/main/scala/geotrellis/store/GeoTrellisRasterSource.scala

```diff
@@ -258,43 +258,16 @@ object GeoTrellisRasterSource {
 
   def readIntersecting(reader: CollectionLayerReader[LayerId], layerId: LayerId, metadata: TileLayerMetadata[_], extent: Extent, bands: Seq[Int], time: Option[ZonedDateTime]): Option[Raster[MultibandTile]] = {
     val tiles = readTiles(reader, layerId, extent, bands, time)
-    sparseStitch(tiles, extent)
+    tiles.sparseStitch(extent)
   }
 
   def read(reader: CollectionLayerReader[LayerId], layerId: LayerId, metadata: TileLayerMetadata[_], extent: Extent, bands: Seq[Int], time: Option[ZonedDateTime]): Option[Raster[MultibandTile]] = {
     val tiles = readTiles(reader, layerId, extent, bands, time)
     metadata.extent.intersection(extent) flatMap { intersectionExtent =>
-      sparseStitch(tiles, intersectionExtent).map(_.crop(intersectionExtent))
+      tiles.sparseStitch(intersectionExtent).map(_.crop(intersectionExtent))
     }
   }
 
-  /**
-    *  The stitch method in gtcore is unable to handle missing spatialkeys correctly.
-    *  This method works around that problem by attempting to infer any missing tiles
-    **/
-  def sparseStitch(
-    tiles: Seq[(SpatialKey, MultibandTile)] with Metadata[TileLayerMetadata[SpatialKey]],
-    extent: Extent
-  ): Option[Raster[MultibandTile]] = {
-    val md = tiles.metadata
```

Comment:
There were no `tiles.isEmpty` check, and anyway we don't need this function anymore, see https://github.com/locationtech/geotrellis/issues/2903

Reply:
Can you comment a bit further on why we don't need this anymore? That linked issue + its links don't show where the original problem `handling missing spatialkeys` is addressed in the original sparsestitch function.

Reply:
@CloudNiner if you will look a bit above (at the place where it was used):

`sparseStitch(tiles, extent)` we can use the inbuild function now `tiles.sparseStitch(extent)`

The issue https://github.com/locationtech/geotrellis/issues/2903 was addressed in your PR here https://github.com/locationtech/geotrellis/pull/3017 which is called "Add sparse stitch method to StitchCollectionMethods".

Reply:
lol embarrassing -- I do in fact remember this now. 👍